### PR TITLE
(PE-17713) Document analytics data collected during update checking.

### PR DESCRIPTION
**Pull request by mindymo**
New OSS documentation from the Product Analytics team clarifying how update checking also sends analytics data to Puppet. 

Please let me know if you have any questions. Thanks!

### CODE REVIEW

**Comment by wkalt** [comment]
@mindymo looks like this needs a rebase against current stable. Content looks fine to me though.

**Comment by mindymo** [comment]
@wkalt Rebase complete!

**Comment by senior** [comment]
:+1: from me, will merge after the release goes out

**Comment by mindymo** [comment]
Thank you! 😃

**Comment by wkalt** [comment]
@mindymo this is meant to correspond to davis right? If so, it actually needs to be re-rebased onto 4.2.x -- we've done some rebranching since this was raised. Sorry for the hassle!

**Comment by mindymo** [comment]
@wkalt I tried to rebase onto 4.2.x, but I'm not totally confident that I've done it correctly. Can you review and let me know if I need to do further battle with Git? Thanks.

File: documentation/configure.markdown [hunk 1]
```diff
@@ -230,9 +230,19 @@ SSL client certificates.
...
+If `disable-update-checking` is set to `false`, PuppetDB checks for updates upon start or restart, and every 24 hours thereafter, and sends the following data to Puppet:
+
+* Product name
+* Database name
+* Database version
+* PuppetDB version
+* IP address
+* Data collection timestamp
+
+The data Puppet collects provides just one of many methods we use for learning about our community of users. The more we know about how you use Puppet, the better we can address your needs. No personally identifiable information is collected, and the data we collect is never used or shared outside Puppet.
```
Comment (wkalt) [inline documentation/configure.markdown]: @mindymo rebase is good, but reading up a bit on "personally identifiable information" led me to wonder whether we want to use those words to exclude IP addresses.

Comment (mindymo) [inline documentation/configure.markdown]: @wkalt From Susannah, who worked with Katie to make sure our data collection practices are OK globally: "We only collect information about companies. We have NO ability to identify individual users whatsoever, and even identification of companies is limited and confined to a small set of our employees. There are no countries who legally designate information about companies as personal information, with the exception of Switzerland."

Comment (wkalt) [inline documentation/configure.markdown]: alright, good enough for me

**Pull request merged**